Sources/Apodini/Semantic Model Builder/SharedSemanticModelBuilder.swift

```diff
@@ -106,27 +106,29 @@ class SharedSemanticModelBuilder: SemanticModelBuilder {
         fatalError("Shared model is unable to deal with .decode")
     }
 
-    static func createRequestHandlerBuilder<C: Component>(with component: C, guards: [LazyGuard] = [], responseModifiers: [() -> (AnyResponseTransformer)] = [])
-                    -> (RequestInjectableDecoder) -> (Vapor.Request) -> EventLoopFuture<Vapor.Response> {
-        { (decoder: RequestInjectableDecoder) in
+    override func encode<T: Encodable>(_ value: T, request: Vapor.Request) throws -> EventLoopFuture<Vapor.Response> {
+        fatalError("Shared model is unable to deal with .encode")
+    }
+
+    static func createRequestHandlerBuilder<C: Component>(with component: C, guards: [LazyGuard] = [], responseModifiers: [() -> (AnyResponseTransformer)] = []) -> (RequestInjectableDecoder & ResponseEncoder) -> (Vapor.Request) -> EventLoopFuture<Vapor.Response> {
+        { (coder: RequestInjectableDecoder & ResponseEncoder) in
             { (request: Vapor.Request) in
                 let guardEventLoopFutures = guards.map { guardClosure in
-                    request.enterRequestContext(with: guardClosure(), using: decoder) { requestGuard in
+                    request.enterRequestContext(with: guardClosure(), using: coder) { requestGuard in
                         requestGuard.executeGuardCheck(on: request)
                     }
                 }
                 return EventLoopFuture<Void>
                         .whenAllSucceed(guardEventLoopFutures, on: request.eventLoop)
                         .flatMap { _ in
-                            request.enterRequestContext(with: component, using: decoder) { component in
-                                var response: ResponseEncodable = component.handle()
-
+                            request.enterRequestContext(with: component, using: coder) { component in
+                                var response: Encodable = component.handle()
                                 for responseTransformer in responseModifiers {
-                                    response = request.enterRequestContext(with: responseTransformer(), using: decoder) { responseTransformer in
+                                    response = request.enterRequestContext(with: responseTransformer(), using: coder) { responseTransformer in
                                         responseTransformer.transform(response: response)
                                     }
                                 }
-                                return response.encodeResponse(for: request)
+                                return try! coder.encode(AnyEncodable(value: response), request: request)
```

Comment:
I kind of thought we already discussed that in the last work meeting. Do we need to do the encoding in the request handler? I think removing this step here and just leaving it as the exporters responsibility would make the most sense? Especially as you are anyways delegating the encoding process to the exporter via the `ResponseEncoder` interface.
The problem I see with the `ResponseEncoder` interface is that the information, where the response data belongs to, is lost. With this approach we can't do HATEOAS.

To illustrate my point:
I've build a experimental (and I really mean **experimental**) rest exporter over at the `rest` branch. Similar to this PR I also move to `Encodable` (as it was proposed by you in #32).
Here is an excerpt from the request handler hooked up to Vapor: 
https://github.com/Apodini/Apodini/blob/e54306dab10b52634c8aae4d1a9acb6048511d5f/Sources/Apodini/Semantic%20Model%20Builder/REST/EndpointHandler.swift#L50-L61

For HATEOS we need to alter the what is encoded into the response, and the information added is dependent on the current endpoint **and** request (injecting query parameters from the request etc...) handled.
Using the `ResponseEncoder` we loose that ability.

I hope I could better explain my points I made in the last work meeting with that code snippet.

Reply:
This looks good. The only problem I see is, that we cannot make the `requestHandlerBuilder` generic.
```
fileprivate var requestHandlerBuilder: (RequestInjectableDecoder & ResponseEncoder) -> (Vapor.Request) -> EventLoopFuture<Vapor.Response>
```

Reply:
> This looks good so far. The only problem I see is, that we cannot make the requestHandlerBuilder generic (because its a closure).

Why would we need a generic? currently they generic in the `encode` method isn't useful anyways, cause you always pass a `AnyEncodable` to `.encode`, so type information is lost always (and even makes it weird because nobody expects a wrapped type here, creating complications if one would use Reflection on the response.

> Further when creating the createRequestHandlerBuilder we don't know anything about the Exporter yet, so we cannot pass the desired Request type.

The exporter is passed as the first argument for the outer closure. Anyways what information do we need from the Request (request used here would be replaced with our own Apodini request type right?)?

Reply:
Nevermind i can do everything we need before i call the handler